src/ethereum_spec_tools/autoapi_patch.py

```diff
@@ -0,0 +1,61 @@
+"""
+The autoapi function used for documentation does not
+fully capture assignment of object variables.
+
+E.g- in the module test.py
+A.x = 10
+B.x = 5
+
+Both instances are treated as `test.x` instead of `test.A.x` & `test.B.x`
+This could incorrectly lead to duplicates.
+
+This tool is uased as a patch.
+"""
+
+from typing import Any
+
+import astroid
+from autoapi.mappers.python import astroid_utils
+
+
+def get_assign_value_patch(node: Any) -> Any:
+    """
+    The get_assign_value function in autoapi.mappers.python.astroid_utils
+    does not handle Attribute assignment properly in that it only considers the
+    attribute name and omits out the information about the parent object
```

Comment:
Would it be possible to upstream a patch?

Reply:
Yes. I do intend to raise a PR at some point. However, the patch itself introduces other kinds of duplicates which we know how to handle in the `specs` but would not know what to do with in a general context. I need to study those things deeper before upstreaming the patch.

Plus, I see that the `autoapi` repo is not updated that frequently. The last PR merge was in Feb with a  few still in queue. So it might take a few months.

I'll remove the `closes` tag from the PR so we can close the issue only after the patch is upstreamed